src/shared-components/segmentedControl/index.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState, useRef, useEffect } from 'react';
+
+import { SegmentsContainer, SegmentItem, Indicator } from './style';
+import { SegmentedControlProps, SegmentItemType } from './types';
+
+const SegmentedControl: React.FC<SegmentedControlProps> = ({
+  segmentItems,
+  initialActiveId = 1,
+  onClick,
+}) => {
+  const initialActiveItem = segmentItems.find(
+    (item: SegmentItemType) => item.id === initialActiveId,
+  );
+  const initialActiveIndex = initialActiveItem
+    ? segmentItems.indexOf(initialActiveItem)
+    : 0;
+
+  const targetRef = useRef<HTMLDivElement>(null);
+  const [activeSegmentId, setActiveSegmentId] = useState(initialActiveId);
+  const [activeSegmentText, setActiveSegmentText] = useState(
```

Comment:
Independent of the changes to working with the `initialActiveIndex` (though things are made easier if we're no longer concerned with that), there's a couple changes here that I think would be beneficial. Here's the code block I came up with (not yet finalized):

```tsx
const getInitialSegmentText = (
  segmentItems: SegmentItemType[],
  initialActiveId: number,
) => {
  const segmentItem = segmentItems.find((item) => item.id === initialActiveId);

  if (!segmentItem) {
    // TODO: How should we handle this if segmentItems is an empty array?
    return segmentItems[0].text;
  }

  return segmentItem.text;
};

const SegmentedControl: React.FC<SegmentedControlProps> = ({
  segmentItems,
  initialActiveId = 1,
  onClick,
}) => {
  const [activeSegmentText, setActiveSegmentText] = useState(() =>
    getInitialSegmentText(segmentItems, initialActiveId),
  );
```

1. We have `initialActiveItem?.text` as the initial value of `activeSegmentText`, which seems like a code smell. I think we should follow the TypeScript compiler's lead on `.find` always potentially returning undefined and handle a few more cases:
    - If the `initialActiveId` we are providing does not match anything in the `segmentItems`, should we default to the first item in `segmentItems`? Are there any scenarios in which the `id`s and expected sort order wouldn't align that we would need to address?
    - What should the behavior of the component be if we pass in an empty array? The component functionality all depends on there being at least one value. Should we raise an error? We should just `return null` at the top of the function if the length is 0? (Or even 1? Does a SegmentedControl make sense with fewer than 2 options?)
1. Less important--but maybe more fun!--is that we can pass a function into `useState` to allow us to do the calculate the initial state only once. ([Here's the article where I read about it](https://www.benmvp.com/blog/four-characters-optimize-react-component/)). Right now, every time the component re-renders (which is any time we select a new option), we're still doing the calculations to find the `initialActiveItem`, even though we only need to do that once. Using a callback in `useState` limits that code execution to just when it is setting the initial state.

Reply:
I read that same article! I'll take a look 👍

Reply:
I streamlined the component by removing the active ID logic in favor of using only the index, as you suggested 👍 

I also added a check to return `null` if the length of `segmentItems` is zero. It doesn't make a lot of sense to have only one item, but I wouldn't rule out the possibility.